docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -107,104 +378,224 @@ public class ProtectedResource {
 
 The default tenant's `OidcConfigurationMetadata` is injected if the endpoint is public.
 
 [[token-claims-roles]]
-=== Token claims and SecurityIdentity roles
+==== Mapping Token claims and SecurityIdentity roles
 
 The way the roles are mapped to the SecurityIdentity roles from the verified tokens is identical to how it is done for the xref:security-oidc-bearer-token-authentication-concept.adoc[Bearer tokens] with the only difference being that https://openid.net/specs/openid-connect-core-1_0.html#IDToken[ID Token] is used as a source of the roles by default.
 
-Note if you use Keycloak then you should set a `microprofile_jwt` client scope for ID token to contain a `groups` claim, please see the https://www.keycloak.org/docs/latest/server_admin/#protocol[Keycloak Server Administration Guide] for more information.
+[NOTE]
+--
+If you use Keycloak then you should set a `microprofile_jwt` client scope for ID token to contain a `groups` claim, please see the https://www.keycloak.org/docs/latest/server_admin/#protocol[Keycloak Server Administration Guide] for more information.
+--
 
-If only the access token contains the roles and this access token is not meant to be propagated to the downstream endpoints then set `quarkus.oidc.roles.source=accesstoken`.
+But depending on your OIDC provider, roles might be stored in the access token or the user info.
+
+If the access token contains the roles and this access token is not meant to be propagated to the downstream endpoints then set `quarkus.oidc.roles.source=accesstoken`.
 
 If UserInfo is the source of the roles then set `quarkus.oidc.roles.source=userinfo`, and if needed, `quarkus.oidc.roles.role-claim-path`.
 
 Additionally, a custom `SecurityIdentityAugmentor` can also be used to add the roles. For more information, see xref:security-customization.adoc#security-identity-customization[SecurityIdentity customization].
 
+=== Ensuring validity of tokens and authentication data
+
+A core part of the authentication process is ensuring the chain of trust and validity of the information.
+This is done by ensuring tokens are trustable.
+
 [[token-verification-introspection]]
-=== Token verification and introspection
+==== Token verification and introspection
 
-Please see xref:security-oidc-bearer-token-authentication-concept.adoc#token-verification-introspection[Token Verification And Introspection] for details about how the tokens are verified and introspected.
+OIDC inherits the bearer authentication token verification and introspection logic described in detail at xref:security-oidc-bearer-token-authentication-concept.adoc#token-verification-introspection[Token Verification And Introspection].
```

Comment:
```suggestion
Verification process of OIDC authorization code flow tokens follows the bearer authentication token verification and introspection logic described in detail at xref:security-oidc-bearer-token-authentication-concept.adoc#token-verification-introspection[Token Verification And Introspection].
```

Reply:
The bearer token verification is very often done to verify the access tokens posted by SPAs which implement OIDC authorization code flow :-). DevServices for Keycloak UI is a good example, it is an SPA which does the code flow, posts the tokens as bearer tokens to Quarkus.
It would be useful indeed to descrive the relationship between OIDC and OAuth2 though, but in a dedicated section, here I feel `inherits` might be a bit confusing. I can look at adding a dedicated section

Reply:
OK. I'd advise aginst going into details of OIDC vs OAuth2, it's not the purpose of that doc.

Reply:
Applied